tools/integration_tests/util/log_parser/json_parser/helpers.go

```diff
@@ -0,0 +1,150 @@
+	if err != nil {
+		return 0, fmt.Errorf("could not parse %s to int64: %v", token, err)
+	}
+	return res, nil
+}
+
+func readFileLineByLine(reader io.Reader) ([]string, error) {
+	content, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+	return strings.Split(string(content), "\n"), nil
+}
+
+func parseReadFileLog(startTimeStampSec, startTimeStampNanos int64, logs []string,
+	structuredLogs map[int64]*StructuredLogEntry) error {
+
+	// Fetch file handle, process id and inode id from the logs.
+	handle, err := parseToInt64(logs[11][:len(logs[11])-1]) //Remove trailing ","
+	if err != nil {
+		return fmt.Errorf("file handle: %v", err)
+	}
+	pid, err := parseToInt64(logs[9][:len(logs[9])-1]) //Remove trailing ","
+	if err != nil {
+		return fmt.Errorf("process id: %v", err)
+	}
+	inodeID, err := parseToInt64(logs[7][:len(logs[7])-1]) //Remove trailing ","
+	if err != nil {
+		return fmt.Errorf("inode id: %v", err)
+	}
+
+	// ReadFile log entries can come multiple times.
+	// Check if log entry exists in the map for file handle.
+	// If log entry doesn't exist, add it to the map.
+	_, ok := structuredLogs[handle]
+	if !ok {
+		structuredLogs[handle] = &StructuredLogEntry{
+			Handle:           handle,
+			StartTimeSeconds: startTimeStampSec,
+			StartTimeNanos:   startTimeStampNanos,
+			ProcessID:        pid,
+			InodeID:          inodeID,
+			Chunks:           []ChunkData{},
+		}
+	}
+
+	return nil
+}
+
+func parseFileCacheLog(startTimeStampSec, startTimeStampNanos int64, logs []string,
+	structuredLogs map[int64]*StructuredLogEntry,
+	opReverseMap map[string]*handleAndChunkIndex) error {
+
+	// Fetch operation id, file handle, size and offset from the logs.
+	opID := logs[0]
+	handle, err := parseToInt64(logs[8][:len(logs[8])-1]) //Remove trailing ","
+	if err != nil {
+		return fmt.Errorf("file handle: %v", err)
+	}
+	size, err := parseToInt64(logs[6])
+	if err != nil {
+		return fmt.Errorf("size: %v", err)
+	}
+	startOffset, err := parseToInt64(logs[4][:len(logs[4])-1]) //Remove trailing ","
+	if err != nil {
+		return fmt.Errorf("start offset: %v", err)
+	}
+
+	// Fetch the log entry for the particular file handle from the structuredLogs map.
+	logEntry, ok := structuredLogs[handle]
+	if !ok {
+		return fmt.Errorf("ReadFile LogEntry for handle %d not found", handle)
+	}
+
+	// For the first file cache log, log entry will not have object and bucket
+	// name, so populate it.
+	if logEntry.ObjectName == "" && logEntry.BucketName == "" {
+		bucketAndObjectName := logs[2][10 : len(logs[2])-1] // Remove prefix "FileCache(" and suffix ","
+		// bucketAndObjectName will be stored in format <bucketName>:/<objectName>
+		logEntry.BucketName = strings.Split(bucketAndObjectName, ":")[0]
+		logEntry.ObjectName = strings.Split(bucketAndObjectName, ":")[1][1:] // Remove prefix "/"
+	}
+
+	// Create chunk data entry and append it to the filecache logs.
+	chunkData := ChunkData{
+		StartTimeSeconds: startTimeStampSec,
```

Comment:
please check/plan if this requires some changes for this extensibility in future -  add error also in this struct in case we want to put checks on errors ?

Reply:
Added the error string. I will update the parsing logic for error strings in a separate PR.

Reply:
I am removing the error string for now as I don't think it will be required and also seeing lint issues if it is unused. I'll add it back if needed. Creating a backlog bug for the same.